pkg/gatewayserver/io/basicstationlns/basicstationlns.go

```diff
@@ -137,11 +140,39 @@ func (s *srv) handleTraffic(c echo.Context) error {
 	if err != nil {
 		return err
 	}
-	ctx = log.NewContextWithField(s.ctx, "gateway_uid", uid)
+
+	auth := c.Request().Header.Get(echo.HeaderAuthorization)
+	if auth != "" {
+		md := metadata.New(map[string]string{
+			"id":            ids.GatewayID,
+			"authorization": fmt.Sprintf("Bearer %s", auth),
+		})
+		if ctxMd, ok := metadata.FromIncomingContext(ctx); ok {
+			md = metadata.Join(ctxMd, md)
+		}
+		ctx = metadata.NewIncomingContext(ctx, md)
+	} else {
+		// Unauthenticated basic station gateways are handled similar to EUI-only udp gateways.
+		ids.EUI, err = messages.GetEUIfromUID(uid)
+		if err != nil {
+			return err
+		}
+		ids.GatewayID = ""
```

Comment:
The `ids` are set premature;

Check the auth first.
- If there's auth, set the IDs and context
- If there's no auth, there can't be a UID. UIDs are strictly reserved for registered identifiers and require primary identifiers to be present. Only the GS itself creates an ID from an EUI if it's not registered, but BS frontend shouldn't care about that. It should also not have a `GetEUIfromUID()`

So my suggestion is that we have a `/traffic/:id` where `id` is an EUI when there's no auth, and it's a UID when there's auth. This also changes and simplifies the discovery handling.

I think I should have caught this in the previous PR, but now that things come together here I see it only now.